integration/spring-data/base/src/main/java/com/blazebit/persistence/spring/data/base/query/AbstractPartTreeBlazePersistenceQuery.java

```diff
@@ -310,12 +312,20 @@ Query createPaginatedQuery(Object[] values, boolean withCount) {
             return binder.bind(jpaQuery);
         }
 
-        protected void processSetting(EntityViewSetting<?, ?> setting, Object[] values) {
+        @SuppressWarnings("unchecked")
+        protected <T> void processSetting(EntityViewSetting<T, ?> setting, Object[] values) {
             for (JpaParameter parameter : parameters.getOptionalParameters()) {
                 String parameterName = parameter.getParameterName();
                 Object parameterValue = values[parameter.getIndex()];
                 setting.addOptionalParameter(parameterName, parameterValue);
             }
+            int entityViewSettingProcessorIndex = parameters.getEntityViewSettingProcessorIndex();
+            if (entityViewSettingProcessorIndex >= 0) {
+                EntityViewSettingProcessor<T> processor = (EntityViewSettingProcessor<T>) values[entityViewSettingProcessorIndex];
+                if (processor != null) {
+                    processor.acceptEntityViewSetting(setting);
```

Comment:
Use `setting = processor.acceptEntityViewSetting(setting);` instead to allow the processor to override the setting.

Reply:
What if I move the processor before applying the optional parameters, so that the processor can override the setting and the optional parameters are then applied to the new setting?

Reply:
I'm a bit confused, isn't it like that currently?

Reply:
Well, it wasn't but now is 😄 I pushed the change already 🙂